3party/opening_hours/opening_hours.hpp

```diff
@@ -700,12 +703,41 @@ enum class RuleState
   Unknown
 };
 
+struct DateHash
+{
+  std::size_t operator()(time_t const & t) const
+  {
+    std::tm tm = *std::localtime(&t);
```

Comment:
You're converting the time to local time (which is a correct approach) assuming the input (holidays set) has the time zone set (which is not true)
We don't have a time zone info for specific region.
We need to fix that
What we have to do is to update our countries_meta.txt and add timezone info for each country and region, add the implementation to generator to embed this data into mwm files and add the implementation to read this data

https://github.com/organicmaps/organicmaps/blob/master/data/countries_meta.txt

https://github.com/organicmaps/organicmaps/blob/master/generator/region_meta.cpp

https://github.com/organicmaps/organicmaps/blob/master/libs/indexer/feature_meta.hpp#L182

Prepare a separate PR with these changes, please